core/store/src/trie/global.rs

```diff
@@ -0,0 +1,48 @@
+use near_primitives::{errors::StorageError, hash::CryptoHash, types::AccountId};
+
+use crate::Trie;
+
+pub struct GlobalShard(Trie);
+
+impl GlobalShard {
+    pub fn new(trie: Trie) -> Self {
+        Self(trie)
+    }
+
+    pub fn trie(self) -> Trie {
+        self.0
+    }
+
+    pub fn get(&self, key: &GlobalTrieKey) -> Result<Option<Vec<u8>>, StorageError> {
+        self.0.get(&key.to_vec())
+    }
+
+    pub fn contains_key(&self, key: &GlobalTrieKey) -> Result<bool, StorageError> {
+        self.0.contains_key(&key.to_vec())
+    }
+}
+
+pub enum GlobalTrieKey {
+    /// Account -> Contract hash
+    Account { account_id: AccountId },
+    /// Contract hash -> Contract code
+    ContractHash { hash: CryptoHash },
+}
```

Comment:
I think I'm missing some context about this, can you share some reference docs about this design?